ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveDruidRules.java

```diff
@@ -54,49 +58,102 @@
  */
 public class HiveDruidRules {
```

Comment:
`HiveDruidRules` can be simplified when we upgrade to Calcite 1.26 onwards (exploiting the work done in https://issues.apache.org/jira/browse/CALCITE-4200). Please add an entry in `org.apache.hadoop.hive.ql.optimizer.calcite.Bug` and link it somehow (e.g., reference it via javadoc) with this class to stand as a reminder.